test/components/tabs.spec.tsx

```diff
@@ -0,0 +1,290 @@
+                driver.tabListPressEnter();
+
+                assertOnChange(onChange, '0');
+            });
+        });
+
+    });
+
+    describe('uncontrolled', () => {
+        describe('defaultValue', () => {
+            it('should render corresponding tab as active', () => {
+                const {driver} = render(
+                    <Tabs defaultValue="1">
+                        <Tab label="Tab One">
+                            <span data-automation-id="FIRST_TAB">Tab One Content</span>
+                        </Tab>
+                        <Tab label="Tab Two">
+                            <span data-automation-id="SECOND_TAB">Tab Two Content</span>
+                        </Tab>
+                    </Tabs>
+                );
+                expect(driver.selectTabContent('FIRST_TAB')).not.to.be.present();
+                expect(driver.selectTabContent('SECOND_TAB')).to.be.present();
+            });
+
+            it('should render the first tab as active when undefined', async () => {
+                const {driver} = render(
+                    <Tabs>
+                        <Tab label="Tab One">
+                            <span data-automation-id="FIRST_TAB">Tab One Content</span>
+                        </Tab>
+                        <Tab label="Tab Two">
+                            <span data-automation-id="SECOND_TAB">Tab Two Content</span>
+                        </Tab>
+                    </Tabs>
+                );
+
+                expect(driver.selectTabContent('FIRST_TAB')).to.be.present();
+                expect(driver.selectTabContent('SECOND_TAB')).not.to.be.present();
+            });
+        });
+
+        it('should change tabs on user interactions', () => {
+            const {driver} = render(
+                <Tabs defaultValue="0">
+                    <Tab label="Tab One">
+                        <span data-automation-id="FIRST_TAB">Tab One Content</span>
+                    </Tab>
+                    <Tab label="Tab Two">
+                        <span data-automation-id="SECOND_TAB">Tab Two Content</span>
+                    </Tab>
+                </Tabs>
+            );
+
+            expect(
+                driver.selectTabContent('FIRST_TAB'),
+                'initial: first tab is active'
+            ).to.be.present();
+            expect(
+                driver.selectTabContent('SECOND_TAB'),
+                'initial: second tab is not active'
+            ).not.to.be.present();
+
+            driver.tabListFocus();
+            driver.tabListKeyDown(KeyCodes.right);
+            driver.tabListPressEnter();
+
+            expect(
+                driver.selectTabContent('FIRST_TAB'),
+                'result: first tab is active'
+            ).not.to.be.present();
+            expect(
+                driver.selectTabContent('SECOND_TAB'),
+                'result: second tab is not active'
+            ).to.be.present();
+        });
```

Comment:
I think you can save result of `driver.selectTabContent('FIRST_TAB')` at least, but in common feels like this test should be smaller

Reply:
To be dead honest we have to query for the tabs at the time of assertions since they get mount/unmount by the component. Also, I would love to see the test shorter, but `SelectionList` requires this number of steps to get its value changed and asserting the exact states of the component helped me a lot.